sound/hda/intel-dsp-config.c

```diff
@@ -414,8 +425,13 @@ static const struct config_entry *snd_intel_dsp_find_config
 			continue;
 		if (table->dmi_table && !dmi_check_system(table->dmi_table))
 			continue;
-		if (table->codec_hid[0] && !acpi_dev_present(table->codec_hid, NULL, -1))
-			continue;
+		if (table->codec_hid) {
+			int i;
+
+			for (i = 0; i < table->codec_hid->num_codecs; i++)
+				if (!acpi_dev_present(table->codec_hid->codecs[i], NULL, -1))
+					continue;
+		}
```

Comment:
Is this snippet equivalent to the previous version? 
In this case the `continue` keyword only skips the inner for loop (the one at line 431) while it probably should "continue" the outer loop, or am I missing something?